tokio/tests/async_send_sync.rs

```diff
@@ -454,6 +454,7 @@ assert_value!(tokio::task::JoinSet<NN>: !Send & !Sync & Unpin);
 assert_value!(tokio::task::JoinSet<YN>: Send & Sync & Unpin);
 assert_value!(tokio::task::JoinSet<YY>: Send & Sync & Unpin);
 assert_value!(tokio::task::LocalSet: !Send & !Sync & Unpin);
+assert_value!(tokio::task::coop::RestoreOnPending: Send & !Sync & Unpin);
```

Comment:
I don't think we want `RestoreOnPending` to be `Send`. That would let you restore the coop state on a different thread ... 

Consider adding a `PhantomData<*mut ()>` field to make it not be `Send`.